test/main/test.js

```diff
@@ -24,6 +24,21 @@ QUnit.test( "basics", function( assert ) {
 	);
 } );
 
+QUnit.test( "setup for null reset", function( assert ) {
+	assert.expect( 0 );
+	QUnit.config.fixture = null;
+	document.getElementById( "qunit-fixture" ).innerHTML = "foobar";
+} );
+
+QUnit.test( "basics for null reset", function( assert ) {
+	assert.equal(
+		document.getElementById( "qunit-fixture" ).innerHTML,
+		"foobar",
+		"did not reset after last test"
+	);
+	QUnit.config.fixture = "test markup";
```

Comment:
This is missing tests for `QUnit.config.fixture = userSpecifiedContent`, and is also a bit ugly (admittedly a preexisting condition). Could we use beforeEach/afterEach to make it more robust and (arguably) a little cleaner?
```js
QUnit.module( "fixture", function( hooks ) {
	var failure = false,
		values = [ /* initial value (see unshift below), */ "<b>ar</b>", undefined ],

		// Use values as a "not-set" sentinel
		originalValue = values;
	hooks.beforeEach( function() {
		if ( originalValue === values ) {
			originalValue = QUnit.config.fixture;
			values.unshift( originalValue );
		}
	} );

	// Set QUnit.config.fixture for the next test, propagating failures to recover the sequence
	hooks.afterEach( function( assert ) {
		if ( failure || values.length === 0 ) {
			assert.ok( false, "prior failure" );
			QUnit.config.fixture = originalValue;
		} else {
			QUnit.config.fixture = values.shift();
		}
	} );

	QUnit.test( "setup", function( assert ) {
		document.getElementById( "qunit-fixture" ).innerHTML = "foo";
		assert.equal( values.length, 3, "proper sequence" );
		failure = failure || values.length !== 3;
	} );
	QUnit.test( "automatically reset", function( assert ) {
		var contents = document.getElementById( "qunit-fixture" ).innerHTML;
		assert.equal( contents, "test markup" );
		assert.equal( values.length, 2, "proper sequence" );
		failure = failure || values.length !== 2 || contents !== "test markup";
	} );
	QUnit.test( "user-specified", function( assert ) {
		var contents = document.getElementById( "qunit-fixture" ).innerHTML;
		document.getElementById( "qunit-fixture" ).innerHTML = "baz";
		assert.equal( contents, "<b>ar</b>" );
		assert.equal( values.length, 1, "proper sequence" );
		failure = failure || values.length !== 1 || contents !== "<b>ar</b>";
	} );
	QUnit.test( "disabled", function( assert ) {
		var contents = document.getElementById( "qunit-fixture" ).innerHTML;
		assert.equal( contents, "baz" );
		assert.equal( values.length, 0, "proper sequence" );
		failure = failure || values.length !== 0 || contents !== "baz";
	} );
} );
```